File: Ruby/Legacy/Condition.rb
#! /usr/bin/env ruby

require_relative "Utility.rb"

def Condition

	test1 = true
	test2 = false

	pnl 1

	println "これから条件分岐を試します",2

	if test1
		println "どうやらtest1は真のようです"
	end

	if not test1
		println "どうやらtest1は偽のようです"
	elsif test2
		println "どうやらtest2は真のようです"
	elsif not test2 and test1
		println "真偽が混在しているようです"
	else
		println "どれでもないようです"
	end

	unless test2
		println "どうやらtest2は偽のようです"
	else
		println "どうやらtest2は真のようです"
	end

	println "test1が真だからこんなことをしているんです" if test1
	println "test2が偽でなければこんなことしないのに" upnless test2

	val = 8
	case val
		when 0,1 then
			println "#{val}は特別な値である"
		when 8..10 then
			println "#{val}は連続して素数ではない"
		when 4,6 then
			println "#{val}は素数ではない"
		when 2,3,5,7 then
			println "#{val}は素数である"
		else
			println "#{val}なぞ私には興味がない"
	end

	pnl 2

end

Condition() if runningDirectly(__FILE__)

File: JavaScript/Legacy/Condition.js
#! /usr/bin/env node

let u=require("./Utility.js");
let e=exports;

e.Condition=()=>{

	let test1 = true;
	let test2 = false;

	u.pnl(1);

	u.println("これから条件分岐を試します",2);

	if (test1) {
		u.println("どうやらtest1は真のようです");
	}

	if (!test1) u.println("どうやらtest1は偽のようです");
	else if (test2) u.println("どうやらtest2は真のようです");
	else if (!test2 && test1) u.println("真偽が混在しているようです");
	else u.println("どれでもないようです");


	let val = 8;
	switch (val) {
		case 0:
		case 1:
			u.println(val+"は特別な値である");break;
		case 8:
		case 9:
		case 10:
			u.println("これに隣接する数にも言えたりするが,");
			// breakを省くと,case 8,9,10を実行すれば,直下のcase 4,6も実行する
		case 4:
		case 6:
			u.println(val+"は素数ではない");break;
		case 2:
		case 3:
		case 5:
		case 7:
			u.println(val+"は素数である");break;
		default:
			u.println(val+"なぞ私には興味がない");break;
	}

	u.pnl(2);

};

if (!module.parent) e.Condition();